### PR TITLE
Update to futures-preview 0.3.0-alpha.16

### PR DESCRIPTION
**Pull request by vorot93**
This includes a switch to new `await` syntax. Warrants a version bump as well.

### CODE REVIEW

**Comment by tikue** [comment]
Thanks so much! Do you know if the `generators` feature is still needed?

File: tarpc/examples/server_calling_server.rs [hunk 1]
```diff
@@ -59,7 +58,7 @@ impl DoubleService for DoubleServer {
 
     fn double(self, _: context::Context, x: i32) -> Self::DoubleFut {
         async fn double(mut client: add::Client, x: i32) -> Result<i32, String> {
-            let result = await!(client.add(context::current(), x, x));
+            let result = client.add(context::current(), x, x).await;
```
Comment (tikue) [inline tarpc/examples/server_calling_server.rs]: No reason for a temp variable anymore; mind inlining this line with the next? It's a good example of await chaining.

Comment (vorot93) [reply tarpc/examples/server_calling_server.rs]: Good catch!

**Comment by vorot93** [comment]
`generators` feature is not necessary any more. Removed it.

**Pull request merged**